### PR TITLE
CD-73975 Service check request should return the response of Sand, instead of just true/false

### PR DESCRIPTION
**Pull request by gubenyas**
* [Main JIRA ticket](https://coupadev.atlassian.net/browse/cd-73975)

reviewers:
- [x] @johnwu96822

### CODE REVIEW

File: service_test.go [hunk 1]
```diff
@@ -45,16 +45,16 @@ var _ = Describe("Service", func() {
...
 					r.Header.Set("Authorization", "")
 					t, err := service.CheckRequest(&r, []string{"scope"}, "")
-					Expect(t).To(Equal(false))
+					Expect(t["allowed"]).To(Equal(false))
 					Expect(err).To(BeNil())
 
 					r.Header.Set("Authorization", "bad bearer token")
 					t, err = service.CheckRequest(&r, []string{"scope"}, "")
-					Expect(t).To(Equal(false))
+					Expect(t).To(Equal(map[string]interface{}{"allowed": false}))
```
Comment (johnwu96822) [inline service_test.go]: You should be able to reference `notAllowedResponse`

File: service_test.go [hunk 2]
```diff
@@ -111,9 +111,9 @@ var _ = Describe("Service", func() {
 
 		Describe("#isTokenAllowed", func() {
 			Context("with empty token", func() {
-				It("returns false", func() {
+				It("returns response with allowed: false", func() {
 					t, err := service.isTokenAllowed("", []string{"scope"}, "", -1)
-					Expect(t).To(Equal(false))
+					Expect(t).To(Equal(map[string]interface{}{"allowed": false}))
```
Comment (johnwu96822) [inline service_test.go]: You should be able to reference `notAllowedResponse`

File: service_test.go [hunk 3]
```diff
@@ -144,13 +144,13 @@ var _ = Describe("Service", func() {
 						fmt.Fprintf(w, string(exp))
 					}
 					t, err := service.isTokenAllowed("abc", []string{"scope"}, "", -1)
-					Expect(t).To(Equal(true))
+					Expect(t).To(Equal(map[string]interface{}{"allowed": true}))
```
Comment (johnwu96822) [inline service_test.go]: You should be able to reference `notAllowedResponse`

**Comment by johnwu96822** [comment]
Please edit the comments for the changed functions and also the README. Thanks!

File: service.go [hunk 4]
```diff
@@ -51,24 +55,24 @@ func NewService(id, secret, tokenURL, resource, verifyURL string, scopes []strin
...
-//CheckRequest checks the bearer token of an incoming HTTP request.
+//CheckRequest checks the bearer token of an incoming HTTP request and return response with 'allowed' true/false field.
 //If the error is of type sand.ConnectionError, the service should respond with
 //HTTP status code 502. Otherwise the client would perform unnecessary retries.
 //Example with Gin:
 //  func(c *gin.Context) {
-//    good, err := sandService.CheckRequest(c.Request, "action", []string{"s1", "s2"})
-//    if err != nil || !good {
-// 	    c.JSON(sandService.ErrorCode(err), err)    //This would set 502 on ConnectionError
+//    response, err := sandService.CheckRequest(c.Request, "action", []string{"s1", "s2"})
```
Comment (johnwu96822) [inline service.go]: Could you swap the action and the scopes parameters? They were in the wrong order.

File: README.md [hunk 5]
```diff
@@ -47,8 +47,8 @@ service.DefaultExpTime = 3600,  # The default expiry time for cache for invalid
 
 //Usage Example with Gin:
 func(c *gin.Context) {
-  good, err := sandService.CheckRequest(c.Request, []string{"scope1", "scope2"}, "action")
-  if err != nil || !good {
+  response, err := sandService.CheckRequest(c.Request, "action", []string{"s1", "s2"})
```
Comment (johnwu96822) [inline README.md]: Could you swap the action and the scopes parameters? They were in the wrong order.

**Comment by gubenyas** [comment]
@johnwu96822 yes, done

**Comment by johnwu96822** [comment]
👍  Thanks!

**Comment by johnwu96822** [comment]
[lockit v3.1.1] Merged via Lockit CLI on 2017-04-13 11:06:05.564690002 -0700 PDT

**Pull request merged**